src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -375,45 +238,146 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
+    if (!apiResponse.isSuccess()) {
+      throw new RuntimeException("Error : " + apiResponse);
+    }
+    return getRecordCountFromHeader(apiResponse);
+  }
+
+  /**
+   * Create a new record in the ServiceNow Table
+   *
+   * @param tableName ServiceNow Table name
+   * @param entity    Details of the Record to be created
+   * @description This function is being used in end-to-end (e2e) tests to fetch a record from the ServiceNow Table.
+   */
+  public String createRecord(String tableName, HttpEntity entity) throws IOException {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      this.conf.getRestApiEndpoint(), tableName, false);
+    String systemID;
+    RestAPIResponse apiResponse = null;
     try {
       String accessToken = getAccessToken();
       requestBuilder.setAuthHeader(accessToken);
-      apiResponse = executeGet(requestBuilder.build());
+      requestBuilder.setAcceptHeader("application/json");
+      requestBuilder.setContentTypeHeader("application/json");
+      requestBuilder.setEntity(entity);
+      apiResponse = executePost(requestBuilder.build());
+
+      systemID = String.valueOf(getSystemId(apiResponse));
+
       if (!apiResponse.isSuccess()) {
         LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
-        collector.addFailure("Unable to fetch schema for table " + tableName, null).
-          withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAME);
-        return null;
-      }
-      List<SchemaResponse> result = parseSchemaResponse(apiResponse.getResponseBody());
-      List<ServiceNowColumn> columns = new ArrayList<>();
-
-      if (result != null && !result.isEmpty()) {
-        for (SchemaResponse field : result) {
-          columns.add(new ServiceNowColumn(field.getName(), field.getInternalType()));
-        }
+      } else {
+        LOG.info(apiResponse.getResponseBody());
       }
-      return SchemaBuilder.constructSchema(tableName, columns);
-    } catch (OAuthSystemException | OAuthProblemException e) {
-      LOG.error("Error in connection - {}", e);
-      collector.addFailure("Connection failed. Unable to fetch schema for table " + tableName, null);
+    } catch (OAuthSystemException | OAuthProblemException | UnsupportedEncodingException e) {
+      LOG.error("Error in creating a new record", e);
+      throw new RuntimeException("Error in creating a new record");
     }
-    return null;
+    return systemID;
   }
 
-  @VisibleForTesting
-  public List<SchemaResponse> parseSchemaResponse(String responseBody) {
-    JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
-    JsonArray ja = jo.getAsJsonArray(ServiceNowConstants.RESULT);
-    Type type = new TypeToken<List<SchemaResponse>>() {
-    }.getType();
-    return gson.fromJson(ja, type);
+  private String getSystemId(RestAPIResponse apiResponse) {
+    JsonObject jsonObject = gson.fromJson(apiResponse.getResponseBody(), JsonObject.class);
```

Comment:
similar comment as above, we shouldn't be doing any json parsing or working directly with JsonObject. There should be a concrete class that represents the response body.

Reply:
Added APIResponse class and deserialized the response directly to the Java object. a963c57e661447a3f1a6e970cdabc6f76964b313